File: LeetCode75-sols/Array_String/Java/Greatest_No_Candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int orig_max = 0;
        for (int j: candies) {
            if (orig_max < j) {
                orig_max = j;
            }
        }
        List<Boolean> result = new ArrayList<>();
        for (int i: candies) {
            if (i + extraCandies >= orig_max) {
                result.add(true);
            }
            else {
                result.add(false);
            }
        }
        return result;
    }
}

File: LeetCode75-sols/Array_String/C/Greatest_No_Candies.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
bool* kidsWithCandies(int* candies, int candiesSize, int extraCandies, int* returnSize) {
    // Find max number of candies a kid has
    int orig_max = candies[0];
    for (int i = 1; i < candiesSize; i++) {
        if (candies[i] > orig_max) {
            orig_max = candies[i];
        }
    }
    
    // Allocate memory for result
    bool* result = (bool*)malloc(candiesSize * sizeof(bool));
    if (result == NULL) {
        *returnSize = 0;
        return NULL;
    }

    // Can each kid have greatest number of candies?
    for (int i = 0; i < candiesSize; i++) {
        if (candies[i] + extraCandies >= orig_max) {
            result[i] = true;
        } else {
            result[i] = false;
        }
    }

    *returnSize = candiesSize;

    return result;
}
